src/main/java/uk/gov/hmcts/reform/ccd/documentam/client/dmstore/DocumentStoreClient.java

```diff
@@ -44,22 +43,21 @@ public DocumentStoreClient(final RestTemplate restTemplate,
         this.applicationParams = applicationParams;
     }
 
-    public Optional<Document> getDocument(final UUID documentId) {
+    public Either<RuntimeException, Document> getDocument(final UUID documentId) {
```

Comment:
could we limit this to ResourceNotFoundException only?

Reply:
The `Either` returns 2 types of exceptions `HttpClientErrorException` or `ResourceNotFoundException` depending on what happens.  The both ultimately inherit `RuntimeException`. 
It could be changed to `Either<Throwable, Document>`?

Reply:
One usage of `getDocument(xxx)` doesn't an throw exception at all when the call returns exceptionally:

```public String generateHashToken(UUID documentId) {
        return documentStoreClient.getDocument(documentId)
            .fold(
                left -> "",
                right -> generateHashToken(documentId, right)
            );
    }
```
As a result, we've got to handle both `HttpClientErrorException` and `ResourceNotFoundException` at the service layer.

Reply:
i think even that code throws HttpClientErrorException if DM is failed other than 404.

Reply:
Got you.  This method should swallow only the 404, but all others propagated.